From cce9a2f130070d799000cac42fe24789d2b777ab Mon Sep 17 00:00:00 2001
From: Sven van Haastregt <sven.vanhaastregt@arm.com>
Date: Wed, 15 Jun 2022 15:46:45 +0100
Subject: [PATCH 58/71] [CMake] Enable LLVM_BUILD_TOOLS for out-of-tree builds

This ensures the llvm-spirv tool is added to the "all" target, so that
it is included when installing the project.

(cherry picked from commit ae904012e2739fa152412e0f1bfc8eb32a808854)
---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 80d99eb2..e7cc1b5c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,11 @@ if (NOT DEFINED LLVM_SPIRV_BUILD_EXTERNAL)
   endif(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
 endif (NOT DEFINED LLVM_SPIRV_BUILD_EXTERNAL)
 
+if(LLVM_SPIRV_BUILD_EXTERNAL)
+  # Make sure llvm-spirv gets built when building outside the llvm tree.
+  set(LLVM_BUILD_TOOLS ON)
+endif(LLVM_SPIRV_BUILD_EXTERNAL)
+
 # Download spirv.hpp from the official SPIRV-Headers repository.
 # One can skip this step by manually setting
 # LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR path.
-- 
2.20.1

